src/main/java/robot/subsystems/shooter/commands/Shoot.java

```diff
@@ -0,0 +1,71 @@
+package robot.subsystems.shooter.commands;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.command.Command;
+
+import static robot.Robot.shooter;
+import static robot.Robot.shooterTable;
+
+/**
+ *
+ */
+public class Shoot extends Command {
+    private double distance;
+    private double timeout;
+    private NetworkTableEntry velocityEntry = shooterTable.getEntry("velocity");
+    private Timer timer = new Timer();
+
+    public Shoot(double distance, double timeout) {
+        requires(shooter);
+        this.distance = distance;
+        this.timeout = timeout;
+    }
+
+    public Shoot(double distance) {
+        this(distance, 0);
+    }
+
+    // Called just before this Command runs the first time
+    @Override
+    protected void initialize() {
+        timer.reset();
+        timer.start();
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    @Override
+    protected void execute() {
+        // shooter.setSpeed((calculateInitialVelocity(distance) / RADIUS) * TICKS_PER_METER); // Convert from linear velocity to radial velocity: v = wr.
+        shooter.setSpeed(calculateVelocity(distance));
+        System.out.println(shooter.getSpeed() + ", calculated speed " + calculateVelocity(distance));
+        setNetworkTable();
+//        System.out.println(shooter.getPosition());
+    }
+
+    private void setNetworkTable() {
+        velocityEntry.setDouble(shooter.getSpeed());
+    }
+
+    private double calculateVelocity(double distance) {
+        return (458.22 * Math.exp(0.2559 * distance)) * (4.0 / 5);
+    }
+
+    // Make this return true when this Command no longer needs to run execute()
+    @Override
+    protected boolean isFinished() {
+        return timeout > 0 && timer.get() >= timeout;
+    }
+
+    // Called once after isFinished returns true
+    @Override
+    protected void end() {
+        timer.stop();
```

Comment:
This line must be redundant. The `timer` object should be destroyed with the command, so the stopping is not needed.